### PR TITLE
CosmosDB: Session token optimization and ChangeFeedProcessor bug fixes

### PR DESCRIPTION
**Pull request by milismsft**
By default we don't scope the global session token in Gateway when partition key ID is part of the request. For containers with a very large number of partitions and multi-region account setting, this can result in "Request headers too long" errors.
The change is to look first at the current request's headers and if the request is scoped to one specific partition, we should also scope the session token to that specific partition. This change is primarily targeting the ChangeFeedProcessor implementation where it's confirmed that we set the partition key ID information as part of the header prior to adding the session token. A more general fix addressing any other cases is not within the scope of these changes.

ChangeFeedProcessor related changes: when the renewer task is cancelled, we also need to close the processor task even if we are about to check-point the current state. The change also ensures that any exceptions are prioritized accordingly.

Added more logs to CFP to allow for better monitoring of the current processing.

Added test case for the scoped session token and simplified the existing CFP test.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionContainer.java [hunk 1]
```diff
@@ -109,6 +109,13 @@ String getSessionToken(String collectionLink) {
     public String resolveGlobalSessionToken(RxDocumentServiceRequest request) {
         ConcurrentHashMap<String, ISessionToken> partitionKeyRangeIdToTokenMap = this.getPartitionKeyRangeIdToTokenMap(request);
         if (partitionKeyRangeIdToTokenMap != null) {
+            String partitionKeyRangeId = request.getHeaders().get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID);
```
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionContainer.java:112]: Is this header present for all crud/query/readFeed ?

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionContainerTest.java [hunk 2]
```diff
@@ -68,6 +68,17 @@ public void sessionContainer() throws Exception {
 
         sessionToken = sessionContainer.resolvePartitionLocalSessionToken(request, resolvedPKRange.getId());
         assertThat(sessionToken.getLSN()).isEqualTo(2);
+
+        String globalSessionToken = sessionContainer.resolveGlobalSessionToken(request);
```
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionContainerTest.java:72]: We have SessionTest.java where we have e2e test and way to intercept the request over the wire. Please all some crud/query operation there to validate gateway/direct mode behaving same

**Review by simplynaveen20 — CHANGES_REQUESTED** [review]
As per our discussion we have issue where gateway mode sending session token for every create request, please cover that issue

**Comment by simplynaveen20** [comment]
As we discussed/found that in gateway mode sdk is sending token unnecessary for every write request, please cover that part as well

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionContainer.java (hunk 1 above)
Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionContainer.java:112]: Only checked for the ReadFeed, more precisely the ChangeFeedProcessor related request where  I could confirm that the header is set prior to adding the session token header.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionContainerTest.java (hunk 2 above)
Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionContainerTest.java:72]: The PR is intended to address primarily the ChangeFeedProcessor case where we have the partition key ID header set as a potential hot fix for users that we recommended to use Gateway mode with CFP. We will need a more broader fix that addresses all the other cases where we send the full session token when we shouldn't.

**Comment by milismsft** [comment]
changes were merged as part of different PRs

**Pull request closed without merging**